### ISSUE TITLE
Make HashOfTag testable without accessing web

### ISSUE THREAD

**Issue by mximp** [objectionary/eo#1174]
We need to be able to populate tags with custom values rather than always download it from the web.

**Comment by yegor256** [comment]
@mximp it's hard to imagine how we can make it testable _alone_, since it's used in `PullMojo` by `new`. Maybe it's better to introduce new option for `PullMojo`, something like: `-DofflineHash=*.*.*:a1b2c3d`. Here, `*.*.*` means "for any version" and `a1b2c3d` is the version that it has to return. Then, we just create a fake implementation of `HashOfTag`, which is created if this `offlineHash` option is provided, and used instead of a regular `HashOfTag`. WDYT?

**Comment by mximp** [comment]
@yegor256 great idea. The only thing I would make this property to contain a list of such mappings: `-DofflineHash="0.2.7:abc2sd3,0.2.8:s4se2fe"`

**Comment by yegor256** [comment]
@mximp yes, a list sounds like a good idea. A file too:

```
"-DofflineHash=0.*.*:abc2sd3"
"-DofflineHash=0.2.7:abc2sd3,0.2.8:s4se2fe"
"-DofflineHashFile=hello.tags"
```

**Comment by yegor256** [comment]
@mximp I suggest assigning this to someone

**Comment by mximp** [comment]
@Nypiaka please assist with the implementation

**Comment by mximp** [comment]
@yasamprom please assist

**Comment by mximp** [comment]
@volodya-lombrozo please take this.

**Comment by 0pdd** [comment]
@mximp the puzzle [#1374](https://github.com/objectionary/eo/issues/1374) is still not solved.

**Comment by 0pdd** [comment]
@mximp 3 puzzles [#1374](https://github.com/objectionary/eo/issues/1374), [#1381](https://github.com/objectionary/eo/issues/1381), [#1382](https://github.com/objectionary/eo/issues/1382) are still not solved.

**Comment by 0pdd** [comment]
@mximp 3 puzzles [#1374](https://github.com/objectionary/eo/issues/1374), [#1381](https://github.com/objectionary/eo/issues/1381), [#1382](https://github.com/objectionary/eo/issues/1382) are still not solved; solved: [#1387](https://github.com/objectionary/eo/issues/1387).

**Comment by 0pdd** [comment]
@mximp 3 puzzles [#1374](https://github.com/objectionary/eo/issues/1374), [#1381](https://github.com/objectionary/eo/issues/1381), [#1397](https://github.com/objectionary/eo/issues/1397) are still not solved; solved: [#1382](https://github.com/objectionary/eo/issues/1382), [#1387](https://github.com/objectionary/eo/issues/1387).

**Comment by 0pdd** [comment]
@mximp 2 puzzles [#1374](https://github.com/objectionary/eo/issues/1374), [#1412](https://github.com/objectionary/eo/issues/1412) are still not solved; solved: [#1381](https://github.com/objectionary/eo/issues/1381), [#1382](https://github.com/objectionary/eo/issues/1382), [#1387](https://github.com/objectionary/eo/issues/1387), [#1397](https://github.com/objectionary/eo/issues/1397).

**Comment by 0pdd** [comment]
@mximp the puzzle [#1374](https://github.com/objectionary/eo/issues/1374) is still not solved; solved: [#1381](https://github.com/objectionary/eo/issues/1381), [#1382](https://github.com/objectionary/eo/issues/1382), [#1387](https://github.com/objectionary/eo/issues/1387), [#1397](https://github.com/objectionary/eo/issues/1397), [#1412](https://github.com/objectionary/eo/issues/1412).

**Comment by 0pdd** [comment]
@mximp all 6 puzzles are solved here: [#1374](https://github.com/objectionary/eo/issues/1374), [#1381](https://github.com/objectionary/eo/issues/1381), [#1382](https://github.com/objectionary/eo/issues/1382), [#1387](https://github.com/objectionary/eo/issues/1387), [#1397](https://github.com/objectionary/eo/issues/1397), [#1412](https://github.com/objectionary/eo/issues/1412).

**Comment by volodya-lombrozo** [comment]
@mximp we can close it
